### DISCUSSION TITLE
Count the groups in Tabs

### DISCUSSION THREAD

**Discussion by M4SSD35TRUCT10N** [fltk-rs/fltk-rs#444] [Q&A] [answered]
Wonder how I can count the groups in a tabs without going through all items?

**Comment by MoAlyousef** [comment, marked as answer]
You can use the WidgetExt::as_group() method:
```rust
let tab = Tabs::new(/*params*/);
/* add groups */
tab.end();
let mut i = 0;
for t in tab.into_iter() { // this consumes the tab, but you can cheaply clone the tab: tab.clone().into_iter()
    if t.as_group().is_some() {
        i += 1;
    }
}
println!("{}", i);
```
The iter is just syntactic sugar for:
```rust
let mut i = 0;
for t in 0..tab.children() {
    if tab.child(t).unwrap().as_group().is_some() {
        i += 1;
    }
}
println!("{}", i);
```
However both technically will have to go through all items.

**Reply by M4SSD35TRUCT10N** [reply]
Okay, that means there is no method/function call to circumvent that. Then I'll stick with my "iteration" solution spiced with your tab.clone().into_iter() suggestion (again fighting the borrow checker). Thank you a lot.
